src/test/java/uk/gov/hmcts/sptribs/caseworker/event/CaseworkerCaseFlagTest.java

```diff
@@ -50,4 +68,124 @@ void shouldNotConfigureCaseFlagIfFeatureFlagFalse() {
             .doesNotContain(CASEWORKER_CASE_FLAG);
     }
 
+    @Test
+    void shouldSuccessfullyAddFlagRepresentative() {
+        //Given
+        final CaseData caseData = caseData();
+        caseData.setNote("This is a test note");
+
+        final CicCase cicCase = new CicCase();
+        final Set<RepresentativeCIC> set = new HashSet<>();
+        set.add(RepresentativeCIC.REPRESENTATIVE);
+        cicCase.setRepresentativeCIC(set);
+        cicCase.setRepresentativeFullName("Jane Doe");
+        cicCase.setNotifyPartyRepresentative(set);
+        caseData.setCicCase(cicCase);
+        final CaseDetails<CaseData, State> updatedCaseDetails = new CaseDetails<>();
+        final CaseDetails<CaseData, State> beforeDetails = new CaseDetails<>();
+        updatedCaseDetails.setData(caseData);
+        updatedCaseDetails.setId(TEST_CASE_ID);
+        updatedCaseDetails.setCreatedDate(LOCAL_DATE_TIME);
+
+        //When
+        AboutToStartOrSubmitResponse<CaseData, State> response =
+            caseworkerCaseFlag.aboutToSubmit(updatedCaseDetails, beforeDetails);
+        SubmittedCallbackResponse submittedCallbackResponse = caseworkerCaseFlag.flagCreated(updatedCaseDetails, beforeDetails);
+
+
+        //Then
+        assertThat(submittedCallbackResponse).isNotNull();
+
+        updatedCaseDetails.setData(caseData);
+        AboutToStartOrSubmitResponse<CaseData, State> response2 =
+            caseworkerCaseFlag.aboutToSubmit(updatedCaseDetails, beforeDetails);
+    }
+
+    @Test
+    void shouldSuccessfullyApplicantAddFlag() {
+        //Given
+        final CaseData caseData = caseData();
+        caseData.setNote("This is a test note");
+        final CicCase cicCase = new CicCase();
+        final Set<ApplicantCIC> set = new HashSet<>();
+        set.add(ApplicantCIC.APPLICANT_CIC);
+        cicCase.setApplicantCIC(set);
+        cicCase.setApplicantFullName("Jane Doe");
+        cicCase.setNotifyPartyApplicant(set);
+        caseData.setCicCase(cicCase);
+        final CaseDetails<CaseData, State> updatedCaseDetails = new CaseDetails<>();
+        final CaseDetails<CaseData, State> beforeDetails = new CaseDetails<>();
+        updatedCaseDetails.setData(caseData);
+        updatedCaseDetails.setId(TEST_CASE_ID);
+        updatedCaseDetails.setCreatedDate(LOCAL_DATE_TIME);
+
+        //When
+        AboutToStartOrSubmitResponse<CaseData, State> response =
+            caseworkerCaseFlag.aboutToSubmit(updatedCaseDetails, beforeDetails);
+        SubmittedCallbackResponse submittedCallbackResponse = caseworkerCaseFlag.flagCreated(updatedCaseDetails, beforeDetails);
+
+
+        //Then
+        assertThat(submittedCallbackResponse).isNotNull();
+
+        updatedCaseDetails.setData(caseData);
+        AboutToStartOrSubmitResponse<CaseData, State> response2 =
+            caseworkerCaseFlag.aboutToSubmit(updatedCaseDetails, beforeDetails);
```

Comment:
No assertions added for the second aboutToSubmit call. (there are many test cases need to be corrected)

Reply:
@sjamihmcts ^^^

Reply:
Updated the tests with more assertions